docs/Learning-Environment-Design-Agents.md

```diff
@@ -14,57 +15,53 @@ action, for example, move the agent in one direction or another. In order to
 agent in a robotic simulator that must provide fine-control of joint torques
 should make its decisions every step of the simulation. On the other hand, an
 agent that only needs to make decisions when certain game or simulation events
-occur, should call `Agent.RequestDecision()` manually.
+occur, such as in a turn-based game, should call `Agent.RequestDecision()` manually.
 
-## Observations
+## Observations and Sensors
 
-To make decisions, an agent must observe its environment in order to infer the
-state of the world. A state observation can take the following forms:
+To make informed decisions, an agent must first make observations of the stete of
+the environment. The observations are collected by Sensors attached to the agent
+GameObject. By default, agents come with a `VectorSensor` which allows them to
+collect floating-point observations into a single array. There are additional
+sensors which can be attached to the agent GameObject which collect their own
+observations, or modify other observations. These are:
 
-* **Vector Observation** — a feature vector consisting of an array of floating
-  point numbers.
-* **Visual Observations** — one or more camera images and/or render textures.
+* `CameraSensor` - Allows image from `Camera` to be used as observation.
+* `RenderTextureSensor` - Allows content of `RenderTexture` to be used as observation.
+* `RayCastSensor` - Allows information from set of ray-casts to be used as observation.
 
-When you use vector observations for an Agent, implement the
-`Agent.CollectObservations(VectorSensor sensor)` method to create the feature vector. When you use
-**Visual Observations**, you only need to identify which Unity Camera objects
-or RenderTextures will provide images and the base Agent class handles the rest.
-You do not need to implement the `CollectObservations(VectorSensor sensor)` method when your Agent
-uses visual observations (unless it also uses vector observations).
+### Vector Observations
 
-### Vector Observation Space: Feature Vectors
+Vector observations are best used for aspects of the environment which are numerical
+and non-visual. The Policy class calls the `CollectObservations(VectorSensor sensor)`
+method of each Agent. Your implementation of this function must call
+`VectorSensor.AddObservation` to add vector observations.
 
-For agents using a continuous state space, you create a feature vector to
-represent the agent's observation at each step of the simulation. The Policy
-class calls the `CollectObservations(VectorSensor sensor)` method of each Agent. Your
-implementation of this function must call `VectorSensor.AddObservation` to add vector
-observations.
-
-The observation must include all the information an agents needs to accomplish
-its task. Without sufficient and relevant information, an agent may learn poorly
+In order for an agent to learn, the total of all observations should include all the
```

Comment:
"the total of all observations"
This is a bit vague. How about "the information an Agent needs to accomplish its task must be given to the Agent as observations"